### ISSUE TITLE
Geany on Fedora 33 freezes 

### ISSUE THREAD

**Issue by vkireyeu** [geany/geany#2665]
Geany constantly freezes on a linux system while typing text or switching between tabs.
Sometimes "Ctrl+S" helps and a typed text appears.
I am using Fedora 33 with Gnome3 and Geany from the repo.

**Comment by elextr** [comment]
This is a recorded message, please always provide the version of Geany, Glib and GTK (see lines near the top of Help->Debug Messages) and the operating system and version you are using.

**Comment by elextr** [comment]
Also are you using Wayland?

**Comment by vkireyeu** [comment]
12:59:32: Geany INFO		: Geany 1.37.1, ru_RU.UTF-8
12:59:32: Geany INFO		: GTK 3.24.23, GLib 2.66.2
12:59:32: Geany INFO		: OS: Fedora 33 (Workstation Edition) ()

> Also are you using Wayland?
I think yes.
I also tried to run Geany with 'primusrun' and 'optirun' to use normal discrete graphics card instead of the integrated (Intel), but it does not help.

**Comment by elextr** [comment]
Make sure _NO_ plugins are enabled too.

**Comment by elextr** [comment]
Possible duplicate of #2658

**Comment by vkireyeu** [comment]
Plug-ins manager shows nothing enabled.

**Comment by vkireyeu** [comment]
(gdb) backtrace
#0  0x00007ffff7bf380f in poll () at /lib64/libc.so.6
Interrupted Geany under gdb just after the freeze several times, got this (always the same result):
```
#1  0x00007ffff6cfad1e in g_main_context_iterate.constprop () at /lib64/libglib-2.0.so.0
#2  0x00007ffff6ca96ab in g_main_loop_run () at /lib64/libglib-2.0.so.0
#3  0x00007ffff75a320d in gtk_main () at /lib64/libgtk-3.so.0
#4  0x00007ffff7d58f9d in main_lib () at /lib64/libgeany.so.0
#5  0x00007ffff7b251e2 in __libc_start_main () at /lib64/libc.so.6
#6  0x00005555555550ae in _start ()
```

And a lot of threads before it, but maybe it's normal behaviour.
```
[New Thread 0x7fffdd12f640 (LWP 82375)]
[New Thread 0x7fffdc909640 (LWP 82376)]
[New Thread 0x7fffbbfff640 (LWP 82378)]
[New Thread 0x7fffbb7fe640 (LWP 82379)]
[New Thread 0x7fffbaffd640 (LWP 82380)]
[New Thread 0x7fffba7fc640 (LWP 82381)]
[Thread 0x7fffde15d640 (LWP 82373) exited]
[Thread 0x7fffde95e640 (LWP 82372) exited]
[Thread 0x7fffdd95c640 (LWP 82374) exited]
[Thread 0x7fffdffff640 (LWP 82326) exited]
[Thread 0x7fffdc909640 (LWP 82376) exited]
[Thread 0x7fffbaffd640 (LWP 82380) exited]
[Thread 0x7fffbb7fe640 (LWP 82379) exited]
[Thread 0x7fffdf7fe640 (LWP 82371) exited]
[Thread 0x7fffbbfff640 (LWP 82378) exited]
[New Thread 0x7fffbbfff640 (LWP 82383)]
[Thread 0x7fffba7fc640 (LWP 82381) exited]
[New Thread 0x7fffba7fc640 (LWP 82387)]
[Thread 0x7fffbbfff640 (LWP 82383) exited]
[Thread 0x7fffba7fc640 (LWP 82387) exited]
```

**Comment by vkireyeu** [comment]
Funny moment: without gdb Geany freezes for seconds on my system, while within gdb freezes are much shorter.

**Comment by elextr** [comment]
The backtrace shows its waiting for something (poll) probably from the Wayland server, what desktop are you using?

But basically it looks (to me) like its nothing Geany can control.

**Comment by elextr** [comment]
Or try `GDK_BACKEND=x11 geany` to not use Wayland.

**Comment by vkireyeu** [comment]
```GDK_BACKEND=x11 geany``` does not help, I use Gnome 3.

**Comment by elextr** [comment]
A question raised elsewhere is what freezes? Is it the whole UI or just the edit window?

**Comment by vkireyeu** [comment]
UI when switching between tabs too.

**Comment by elextr** [comment]
> And actually I use Geany on Wayland quite a long time, so, problem appears only after Geany 1.37.

But the upgrade to Fedora 33 would have changed a lot more than just Geany, or are you saying Geany 1.36 works fine on Fedora 33 with wayland, but 1.37 doesn't?

**Comment by vkireyeu** [comment]
Actually I do not remember now.
But for sure I used Geany 1.36 (and 1.35) on Fedora 32 together with Wayland and Gnome3.

**Comment by elextr** [comment]
I don't suppose you can try Geany 1.36 on Fedora 33? 

Also thank you for being helpful, I did contact a user on IRC who I thought used Geany on Wayland, but they don't so its not a configuration anybody else has (yet) AFAIK and your input is the only available data we have.

**Comment by vkireyeu** [comment]
There still Geany 1.36 in the repository, so I installed it.
Geany 1.36 + Fedora 33 (Wayland, Gnome3) = freezes.
Geany 1.36 (with ```GDK_BACKEND=x11```) + Fedora 33 (Wayland, Gnome3) = freezes.

**Comment by elextr** [comment]
Ok, so it appears that something that changed in Fedora from 32 to 33 is the problem, possibly GTK versions.  Not sure how to take it further.

**Comment by codebrainz** [comment]
It's probably still Wayland, I don't think you can use per-app environment variables if you're within Wayland. Try to log out and change the session to an X-session instead of Wayland (gear icon -> 'GNOME on Xorg').

**Comment by elextr** [comment]
@vkireyeu try `GDK_BACKEND=help geany` and it should print a list of backends the GTK supports (IIUC) maybe Fedora simply hasn't compiled GTK to support X11 any more.

**Comment by vkireyeu** [comment]
I tried to find how to enable X11 on Fedora or check what I am using (X11 or wayland), so, it seems like the whole time on my laptop I was using X11. Maybe it was because my cards (Intel + Nvidia).

Now I configured system to finally use Wayland (I hope!), so
```
$ echo $XDG_SESSION_TYPE
wayland
```
And... No freezes in Geany!

So, summary:
1) Fedora 33 + **X11** = freezes in the Geany
2) Fedora 33 + **Wayland** = smooth running of the Geany

I am sorry to confuse you.

**Comment by codebrainz** [comment]
@vkireyeu good news!

If you are willing and able, it would be interesting to see if you experience the same issue with [Scite](https://www.scintilla.org/SciTE.html) which is the reference editor for the source code editing widget/component Geany uses (Scintilla). If you experience the same issue with Scite, it would show that there's probably not an issue with Geany's code and the bug could be worked on upstream in Scintilla.

**Comment by vkireyeu** [comment]
It seems like Scite is not in the Fedora repositories, so unfortunately I can't.

**Comment by Orionic47** [comment]
I get the same bugs as @vkireyeu, namely Geany often freezes when typing or erasing text and freezes when attempting to switch tabs. When typing or erasing text a "CTRL+S" sometimes unfreezes Geany. You can type when it is frozen and the input correctly appears on screen when Geany unfreezes. The "CRTL+S" method does not unfreeze Geany when trying to switch tabs. A "ALT+TAB" to switch to other applications (tried with Chrome and Terminal) solves the freeze every time for both the typing freeze and the tab switching freeze.

I am currently using POP!_os 20.10 and Geany 1.36, no plug-ins are enabled. `echo $XDG_SESSION_TYPE` returns `x11`

**Comment by numen3** [comment]
Hello everyone.
I just installed fedora 33 yesterday. I am using the `x11 session`, and the kernel `5.9.13-200.fc33.x86_64`. The system is basically vanilla, I only added the `rpm fusion` repos.

I am experiencing the freezes too. All my addons are disabled and at the startup there is only the blank `untitled` tab document. How to reproduce:
- Open program
- Start typing in the untitled document: some of the times it will freeze, not always though. It seems related to how soon I start typing after the new document opens
- If it doesn't freeze, open a new untitled document with `ctrl+n` and start typing, it will then freeze most of the time. Not always. It seems related to how soon I start typing after the new document opens

It is true that interacting with the window manager makes the text appear (so it make it unfreeze): for instance in some cases the text never shows up until I press `alt+tab`
It is also true that changing tab works for unfreezing.

**Comment by numen3** [comment]
I have tried geany using the wayland session (simply the `gnome` session) at login. It seems that this specific issue is gone, so that it works in the wayland session.

**Comment by numen3** [comment]
One last bit of information which I think may be very important: the problem only occurs if the window is maximized. If not, I cannot manage to reproduce the freeze. There may be some other ways, but, considering the short amount of time I tried, I couldn't make it freeze if the window was unmaximized.

**Comment by numen3** [comment]
I was trying the sublime text editor, and I managed to get it to freeze in maximized mode. To unfreeze I had to do `alt+tab` again.
So this is probably a bug in the distro or in something related to X11/Wayland.

I would use the wayland session, but for me is laggy and some software is still not compatible (rofi doesn't work for me).

**Comment by elextr** [comment]
Since the problem doesn't seem to occur on any other systems except Fedora 33 X11/Wayland it suggests there is a problem in that system.  IIUC Fedora uses XWayland which is an X11 frontend for Wayland rather than Xorg X11 itself, so probably there are bugs in that.

Basically doesn't seem to be anything Geany can do to help except say "use wayland directly" which seems the only workaround.

**Comment by beaumanvienna** [comment]
@elextr @vkireyeu @numen3 @Orionic47 @codebrainz 
I think #2658 is a duplicate. Could you guys explain the workaround for me again, please? Thanks!

**Comment by jansalleine** [comment]
I don't know if this comment will be displayed on a closed issue, but I had the same problem on Arch Linux, Gnome 3, X11 Session. In this case my workaround was to edit /usr/share/applications/geany.desktop and change the "Exec" line to:
```Exec=env GDK_BACKEND=x11 geany %F```
